server/src/main/java/org/opensearch/index/engine/Engine.java

```diff
@@ -175,6 +175,19 @@ public final EngineConfig config() {
      */
     protected abstract SegmentInfos getLatestSegmentInfos();
 
+    /**
+     * Fetch a snapshot of the latest SegmentInfos from the engine. Using this method
```

Comment:
This is not entirely true for the default implementation used by ReadOnly and NRTReplicationEngine where we don't have an indexWriter to incRefDeleter segments.  I think we should call out that this has no impact for these read only engines.

Reply:
Updated the javadocs